src/readwritesb.F90

```diff
@@ -0,0 +1,598 @@
+!> @file
+!> @brief Read or write a BUFR data subset.
+!>
+!> @author J. Woollen @date 1994-01-06
+
+!> Read the next data subset from a BUFR message.
+!>
+!> Logical unit lunit should have already been opened for
+!> input operations via a previous call to subroutine openbf(), and a
+!> BUFR message should have already been read into internal arrays via
+!> a previous call to one of the
+!> [message-reading subroutines](@ref hierarchy).
+!>
+!> Whenever this subroutine returns with iret = 0, this indicates that a
+!> new BUFR data subset (i.e. report) was successfully read into internal
+!> arrays within the NCEPLIBS-bufr software, and from where it can be
+!> manipulated or further parsed via calls to any of the [values-reading
+!> subroutines](@ref hierarchy).
+!>
+!> If the subroutine returns with iret = -1 there are no more data
+!> subsets available within the current message; a new call needs to
+!> be made to one of the [message-reading subroutines](@ref hierarchy)
+!> in order to read in the next message from logical unit lunit.
+!>
+!> @param[in] lunit - integer: Fortran logical unit number for BUFR file
```

Comment:
One thing you might notice is that with F90 code, doxygen can find the types of parameters. That is, you no longer need to have:
`@param[in] lunit - integer: Fortran logical unit number for BUFR file
`
You could switch to:
`@param[in] lunit - Fortran logical unit number for BUFR file
`

Reply:
OK, I didn't know that.  But if that's the case then I'll start leaving those extra descriptions out to avoid redundancy - thanks!